htmd/smallmol/smallmol.py

```diff
@@ -1,105 +1,654 @@
+                  'coords': (0, 3, 1),
+                  '_chiraltags':(0,)
+                    }
+
+    _mol_dims = {'_mol': (0,),
+                 'ligname': (0,),
+                 'totalcharge': (0,)
+                   }
+
+    def __init__(self, mol, ignore_errors=False, force_reading=False, fixHs=True, removeHs=False):
+
+
+        self.mol_fields = self._mol_fields.copy()
+        self.atom_fields = self._atom_fields.copy()
+
+        # init the __dict__ with the fields
+        for field in self._atom_dtypes:
+            self.__dict__[field] = np.empty( self._atom_dims[field], dtype=self._atom_dtypes[field] )
+        for field in self._mol_dtypes:
+            self.__dict__[field] = None
+
+        # load the input and store the rdkit Mol obj
+        # self._mol is used for debugging. Later on will be removed
+        self._mol, smallMol = self._initializeMolObj(mol, force_reading)
+        if not ignore_errors and self._mol == None:
+            if not force_reading:
+                raise ValueError("Unkown '{}' provided. Not a valid mol2,pdb,smile, rdkitMol obj. "
+                                 "Try by setting the force_reading option as True.".format(mol))
+            else:
+                raise ValueError("Unkown '{}' provided. Not a valid mol2,pdb,smile, rdkitMol obj.".format(mol))
+
+        if removeHs:
+            self._mol = Chem.RemoveHs(self._mol)
+
+        if fixHs:
+            self._mol = Chem.AddHs(self._mol, addCoords=True)
+
+        # fill the molecule and atom properties
+        self._readMol(smallMol)
+
+    @property
+    def listProps(self):
+        """
+        The list of properties of the molecule
+
+        Returns
+        -------
+        fields: list
+            The list of properties
+        """
+        _fields = list(self.mol_fields)
+        return _fields
+
+    @property
+    def listPropsAtoms(self):
+        """
+        The list of properties of the atoms
+
+        Returns
+        -------
+        fields: list
+            The list of properties
+        """
+        _fields = list(self.atom_fields)
+        return _fields
 
-    def __init__(self, mol, ignore_errors=False, addHs=True):
+    @property
+    def numAtoms(self):
         """
-        Initializes small molecule object
+        Returns the number of atoms in the molecule
+
+        Returns
+        -------
+        natoms: int
+            The number of atoms in the molecule
+
+        """
+        # getting the number of atoms from the shape of element array
+        natoms = self.element.shape[0]
+        return natoms
+
+
+    @property
+    def numConformers(self):
+        """
+        Returns the number of conformers in the rdkit molecule object
+
+        Returns
+        -------
+        nconfs: int
+            The number of conformers in the molecule
+        """
+        # getting the number of conformers from the shape of coords array
+        _nConformers = self.coords.shape[-1]
```

Comment:
detail: I'd prefer it returning `self.coords.shape[2]`. That way if for some reason you messed up and `self.coords` is two dimensional it will crash instead of returning `3` for `xyz`

Reply:
I see. thanks for the suggestion. Done it